src/components/SubNav/SubNav.tsx

```diff
@@ -287,20 +235,23 @@ export const SubNav: ChakraComponent<
               >
                 {primaryActions}
                 {showRightFade && (
-                  // Explicitly cast styles.fadeEffect to CSSProperties
-                  <div style={styles.fadeEffect as CSSProperties} />
+                  <li>
+                    <Box sx={styles.fadeEffect} />
+                  </li>
```

Comment:
This is not working well. The fade element is moving with the scroll. Also, the action items are flickering when scrolling. Can you wrap the `List` in a `div` and then move the `Box` up one level so it is next to the `List`? That would be more in alignment with how we are dong this for the `Table` component.

I think you will also need to move the styles that power the scrolling.